src/xcode/ENA/ENA/Source/Scenes/Events/Store/Model/Checkin.swift

```diff
@@ -10,12 +10,12 @@ struct Checkin {
 	let traceLocationType: TraceLocationType
 	let traceLocationDescription: String
 	let traceLocationAddress: String
-	let traceLocationStart: Date
-	let traceLocationEnd: Date
-	let traceLocationDefaultCheckInLengthInMinutes: Int
+	let traceLocationStartDate: Date?
+	let traceLocationEndDate: Date?
+	let traceLocationDefaultCheckInLengthInMinutes: Int?
 	let traceLocationSignature: String
 	let checkinStartDate: Date
-	let checkinEndDate: Date
-	let targetCheckinEndDate: Date
+	let checkinEndDate: Date?
+	let targetCheckinEndDate: Date?
```

Comment:
If I understand it correctly, this field should not be optional.

At least according to [EXPOSUREAPP-5582](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5582): "Automatic check-out time is mandatory and must be later than current time"

Reply:
According to the tech spec it is optional. https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/event-registration-mvp/docs/spec/event-registration-client.md#database-table-for-checkins

Reply:
I ask max about that.